lib/Backend/arm64/LowerMD.cpp

```diff
@@ -1611,78 +1690,104 @@ LowererMD::LowerExitInstr(IR::ExitInstr * exitInstr)
-            IR::IndirOpnd::New(
-                IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), (int32)0, TyMachReg, this->m_func),
-            this->m_func);
-        exitInstr->InsertBefore(instrPop);
+        IR::Instr * instrStp = IR::Instr::New(Js::OpCode::LDP_POST, this->m_func);
+        instrStp->SetSrc1(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), (2 * MachRegInt), TyMachReg, this->m_func));
+        instrStp->SetDst(IR::RegOpnd::New(nullptr, EH_STACK_SAVE_REG, TyMachReg, this->m_func));
+        instrStp->SetSrc2(IR::RegOpnd::New(nullptr, SCRATCH_REG, TyMachReg, this->m_func));
+        exitInstr->InsertBefore(instrStp);
 
         LowererMD::CreateAssign(
             IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func),
             IR::RegOpnd::New(nullptr, EH_STACK_SAVE_REG, TyMachReg, this->m_func),
             exitInstr);
     }
 
-    int32 regPopedSize = 0;
-
-    // 2. Restore saved double registers. Generate vpop {d8-d15}
+    // 2. Restore saved double registers.
     BVUnit savedDoubleRegs(this->m_func->m_unwindInfo.GetDoubleSavedRegList());
     if (!savedDoubleRegs.IsEmpty())
     {
-        // ToDo (SaAgarwa): All LDR/STR should be converted to LDP/STP
+        Assert(savedDoubleRegs.Count() % 2 == 0);
+        IR::RegOpnd* reg1Opnd = nullptr;
+        IR::RegOpnd* reg2Opnd = nullptr;
         for (RegNum reg = FIRST_CALLEE_SAVED_DBL_REG; reg <= LAST_CALLEE_SAVED_DBL_REG; reg = (RegNum)(reg + 1))
         {
             Assert(LinearScan::IsCalleeSaved(reg));
             Assert(reg != RegLR);
             if (savedDoubleRegs.Test(RegEncode[reg]))
             {
-                regPopedSize += MachRegDouble;
-                IR::Instr * instrStore = IR::Instr::New(Js::OpCode::FLDR, this->m_func);
-                instrStore->SetSrc1(IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), regPopedSize, TyMachReg, this->m_func));
-                instrStore->SetDst(IR::RegOpnd::New(reg, TyMachReg, this->m_func));
-                exitInstr->InsertBefore(instrStore);
+                if (reg1Opnd == nullptr)
+                {
+                    reg1Opnd = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+                else
+                {
+                    Assert(reg2Opnd == nullptr);
+                    reg2Opnd = IR::RegOpnd::New(reg, TyMachReg, this->m_func);
+                }
+            }
+
+            if (reg1Opnd != nullptr && reg2Opnd != nullptr)
+            {
+                IR::Instr * instrLoadRegister = IR::Instr::New(Js::OpCode::LDP_POST,
+                    reg1Opnd,
+                    IR::IndirOpnd::New(IR::RegOpnd::New(nullptr, RegSP, TyMachReg, this->m_func), 2 * MachRegDouble, TyMachReg, this->m_func),
+                    reg2Opnd,
+                    this->m_func);
+                exitInstr->InsertBefore(instrLoadRegister);
+                reg1Opnd = reg2Opnd = nullptr;
             }
```

Comment:
Mirror comments from above. You can't incrementally load and pop. You must do all the loads with LDP first then release the stack all at once at the end.